lib/shelley/test/unit/Cardano/Wallet/Shelley/NetworkSpec.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE NumericUnderscores #-}
+
+module Cardano.Wallet.Shelley.NetworkSpec (spec) where
+
+import Prelude
+
+import Cardano.BM.Data.Severity
+    ( Severity (..) )
+import Cardano.BM.Trace
+    ( nullTracer )
+import Cardano.Wallet.Primitive.Types
+    ( NetworkParameters (..) )
+import Cardano.Wallet.Shelley.Compatibility
+    ( NodeVersionData )
+import Cardano.Wallet.Shelley.Launch
+    ( singleNodeParams, withBFTNode, withSystemTempDir )
+import Cardano.Wallet.Shelley.Network
+    ( withNetworkLayer )
+import Control.Concurrent
+    ( threadDelay )
+import Control.Concurrent.Async
+    ( replicateConcurrently_ )
+import Test.Hspec
+    ( Spec, describe, it )
+import Test.Utils.Trace
+    ( withLogging )
+
+{-------------------------------------------------------------------------------
+                                      Spec
+-------------------------------------------------------------------------------}
+
+spec :: Spec
+spec = describe "NetworkLayer regression test #1708" $ do
+    it "Parallel local socket connections" $
+        withTestNode $ \np sock vData -> withLogging $ \(tr, _getLogs) ->
+            replicateConcurrently_ 5 $
```

Comment:
Was this enough to observe the failure? (I assume so)

Reply:
Yes - probably 3 concurrently would be enough, but why not 5.